src/System.Numerics.Tensors/System/Numerics/Tensor.cs

```diff
@@ -956,9 +947,16 @@ private IEnumerable<T> Enumerate()
 
         void ICollection.CopyTo(Array array, int index)
         {
-            for(int i = 0; i < length; i++)
+            if (array is T[] destinationArray)
```

Comment:
What is the difference between System.Array and T[]? When is array not a T[]?

Reply:
The interface ICollection specifies that the parameter is `Array`, which can be any array. Someone could pass an `int[]` into a `Tensor<double>`.  Or even `Array` could be a multi-dimension array - `int[,]`.

As @KrzysztofCwalina points out, I need to add parameter validation here.